core/src/main/java/hudson/tools/ToolInstaller.java

```diff
@@ -129,4 +129,24 @@ private String sanitize(String s) {
     public ToolInstallerDescriptor<?> getDescriptor() {
         return (ToolInstallerDescriptor) Jenkins.getInstance().getDescriptorOrDie(getClass());
     }
+
+    public static final class ToolInstallerList {
+        public ToolInstallerEntry [] list;
+    }
+
+    public static final class ToolInstallerEntry {
+        public String id;
+        public String name;
+        public String url;
+
+        public ToolInstallerEntry() {
```

Comment:
yes was throwing an exception on the missing constructor during the JSON creation cannot recall exactly ATM